src/server/req_quejob.c

```diff
@@ -3704,4 +3710,67 @@ validate_place_req_of_job_in_reservation(job *pj)
 
 	return 1;
 }
+
+/**
+ * @brief
+ * 		Provides the next job id
+ *
+ * @param[in] void
+ *
+ * @return	long long
+ * @retval (>=0 to <= max_job_sequence_id): Success
+ * @retval	-1	: database error
+ *
+ */
+
+static
+long long get_next_svr_sequence_id(void) {
+
+	long long ret_svr_sequence_id = 0;
+
+	/* JobId's save to the database after every 1000 jobs */
+	if(svr_sequence_window_count == 0) {
+		/* Save the job-id numbers in the database by a SEQ_WIN_INCR(1000) in advance.
+		   So this way, we have to save it once only after every SEQ_WIN_INCR(1000) submissions.*/
+		long long backup_svr_sequence_id;
+		update_svr_sv_jobidnumber = 0;
+		backup_svr_sequence_id = server.sv_qs.sv_jobidnumber;
```

Comment:
Why do we need this backup_svr_sequence_id? I think this constant swapping of variables is making the code complex and difficult to maintain correctly. All you need is a in-memory variable that keeps track of the current jobid counter, and the server.sv_qs.sv_jobidnumber can the 1000 incremented forward id